src/Logging/V2/ConfigServiceV2Api.php

```diff
@@ -0,0 +1,601 @@
+     *           not retry.
+     *     @type int $timeoutMillis The timeout in milliseconds to use for calls
+     *                              that don't use retries. For calls that use retries,
+     *                              set the timeout in RetryOptions.
+     *                              Default: 30000 (30 seconds)
+     *     @type string $appName The codename of the calling service. Default 'gax'.
+     *     @type string $appVersion The version of the calling service.
+     *                              Default: the current version of GAX.
+     *     @type Google\Auth\CredentialsLoader $credentialsLoader
+     *                              A CredentialsLoader object created using the
+     *                              Google\Auth library.
+     * }
+     */
+    public function __construct($options = [])
+    {
+        $defaultScopes = [
+            'https://www.googleapis.com/auth/cloud-platform',
+            'https://www.googleapis.com/auth/cloud-platform.read-only',
+            'https://www.googleapis.com/auth/logging.admin',
+            'https://www.googleapis.com/auth/logging.read',
+            'https://www.googleapis.com/auth/logging.write',
+        ];
+        $defaultOptions = [
+            'serviceAddress' => self::SERVICE_ADDRESS,
+            'port' => self::DEFAULT_SERVICE_PORT,
+            'scopes' => $defaultScopes,
+            'retryingOverride' => null,
+            'timeoutMillis' => self::DEFAULT_TIMEOUT_MILLIS,
+            'appName' => 'gax',
+            'appVersion' => self::_GAX_VERSION,
+            'credentialsLoader' => null,
+        ];
+        $options = array_merge($defaultOptions, $options);
+
+        $headerDescriptor = new AgentHeaderDescriptor([
+            'clientName' => $options['appName'],
+            'clientVersion' => $options['appVersion'],
+            'codeGenName' => self::_CODEGEN_NAME,
+            'codeGenVersion' => self::_CODEGEN_VERSION,
+            'gaxVersion' => self::_GAX_VERSION,
+            'phpVersion' => phpversion(),
+        ]);
+
+        $defaultDescriptors = ['headerDescriptor' => $headerDescriptor];
+        $this->descriptors = [
+            'listSinks' => $defaultDescriptors,
+            'getSink' => $defaultDescriptors,
+            'createSink' => $defaultDescriptors,
+            'updateSink' => $defaultDescriptors,
+            'deleteSink' => $defaultDescriptors,
+        ];
+        $pageStreamingDescriptors = self::getPageStreamingDescriptors();
+        foreach ($pageStreamingDescriptors as $method => $pageStreamingDescriptor) {
+            $this->descriptors[$method]['pageStreamingDescriptor'] = $pageStreamingDescriptor;
+        }
+
+        // TODO load the client config in a more package-friendly way
+        // https://github.com/googleapis/toolkit/issues/332
+        $clientConfigJsonString = file_get_contents('./resources/config_service_v2_client_config.json');
```

Comment:
We will need to update these to something like:

`file_get_contents(__DIR__ . '/resources/config_service_v2_client_config.json');`